projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/order-history/b2b-order-history-orders-flow.e2e-spec.ts

```diff
@@ -81,4 +85,61 @@ describe('Order History with orders', () => {
     cy.get('.cx-order-history-po a').should('contain', poNumber);
     cy.get('.cx-order-history-cost-center a').should('contain', costCenter);
   });
+
+  describe('Order details - reorder', () => {
+    it('should display order details page with the reorder button', () => {
+      cy.get('.cx-order-history-value').first().click();
+      cy.get('cx-order-details-reorder div div button').should(
+        'contain',
+        'Reorder'
+      );
+    });
+
+    it('items in the cart should match previous order when proceeding with reorder', () => {
+      // Saving cartId before starting the reorder flow
+      b2bCheckout.addB2bProductToCart();
+      cart.goToCart();
+      cart.saveCartId();
+
+      goToOrderDetails();
+      saveOrderDetails();
+
+      // Check if the reorder button exists
+      cy.get('cx-order-details-reorder div div button').first().click();
+      cy.get('.cx-reorder-dialog-areyousure-section').should('exist');
```

Comment:
The test should probably wait for the reorder request to finish before proceeding to mitigate a source of potential flakiness.